src/query/execute_statement.c

```diff
@@ -14809,12 +14809,67 @@ do_replicate_statement (PARSER_CONTEXT * parser, PT_NODE * statement)
     {
       PT_PRINT_VALUE_FUNC saved_func = parser->print_db_value;
       int saved_custom_print = parser->custom_print;
+      PARSER_VARCHAR **host_val;
```

Comment:
How about introducing a new function that performs this processing?

Reply:
how about the below comments?
     /*
       * if the query string includes the host variables, while processing the variable holder '?'
       * the values of the host variables can be replaced into the user's original query string
       * the pt_print_db_value(...) returns the value string and its length.
       * the length includes quotes in case of the char string.
       */

Reply:
I prefer making it a function, but I think the decision is up to you.